data-prepper-plugins/opensearch/src/integrationTest/java/org/opensearch/dataprepper/plugins/sink/opensearch/OpenSearchSinkIT.java

```diff
@@ -1116,6 +1119,61 @@ public void testOpenSearchRoutingField(final String testRoutingField) throws IOE
         sink.shutdown();
     }
 
+    @ParameterizedTest
+    @ValueSource(strings = {"info/ids/rid", "rid"})
+    public void testOpenSearchRoutingFieldWithExpressions(final String testRoutingField) throws IOException, InterruptedException {
+        final String expectedRoutingField = UUID.randomUUID().toString();
+        final String testIndexAlias = "test_index";
+        final Event testEvent = JacksonEvent.builder()
+                .withData(Map.of("arbitrary_data", UUID.randomUUID().toString()))
+                .withEventType("event")
+                .build();
+        testEvent.put(testRoutingField, expectedRoutingField);
+
+        final List<Record<Event>> testRecords = Collections.singletonList(new Record<>(testEvent));
+
+        final PluginSetting pluginSetting = generatePluginSetting(null, testIndexAlias, null);
+        pluginSetting.getSettings().put(IndexConfiguration.ROUTING_FIELD, "${/"+testRoutingField+"}");
+        when(expressionEvaluator.isValidFormatExpression(any(String.class))).thenReturn(true);
```

Comment:
Since we mock this, can you just run one test manually to make sure that 

`routing_field`: "normalField"

does not have any issues just to make sure there is no breaking change. I think it is preferable to deprecate `routing_field` like we did for `document_id_field`, and to have a new `routing_id` that supports the expressions. What do you think?

Reply:
The previous test is with `routing_field` set to normal value. There is no need to test it manually. We can support the `routing_field` to support both normal field and expressions. Isn't that better than introducing `routing_id` and deprecating the current one?

Reply:
Yes it is but we are mocking the expression evaluator so it's really just to make sure that this gets hit (https://github.com/opensearch-project/data-prepper/blob/f19de03d5418925935e019837fc4824fb250820c/data-prepper-expression/src/main/java/org/opensearch/dataprepper/expression/GenericExpressionEvaluator.java#L71) and we don't return it as valid. I just noticed we don't have a test for normal keys here to return false (https://github.com/opensearch-project/data-prepper/blame/f19de03d5418925935e019837fc4824fb250820c/data-prepper-expression/src/test/java/org/opensearch/dataprepper/expression/GenericExpressionEvaluatorTest.java#L134). Can you add a parameter to this test for a regular key

Also the only reason to change would be naming, since `routing_field` implies just a single field to be used, but I guess it isn't that big of a difference

Reply:
I tend to prefer making a new property with this configuration - `routing`. It will be consistent with `document_id` and clarify that you are not supply the field only.

The OpenSearch API uses the term `routing`, so I recommend that over `routing_id`.

Reply:
My only thought about `routing` is that it could collide with `routes`, but I suppose `routing_id` could collide just as much